### PR TITLE
Implement Seq rotateLeft/rotateRight methods

### LINKED ISSUE #1863: Shift / rotate
I missed an obvious method here: https://github.com/javaslang/javaslang/issues/1862... so I'll just ask a question here instead of proposing something!

Is there a `rotate` or `shift` method? 

Thanks.

### PR DESCRIPTION
**Pull request by valery1707**
Fix #1863

### CODE REVIEW

File: vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java [hunk 1]
```diff
@@ -1489,6 +1489,86 @@ public void shouldCreateReverseIteratorOfNonEmpty() {
         assertThat(ofAll(of("a", "b", "c")).reverseIterator().toList()).isEqualTo(Iterator.of("c", "b", "a").toList());
     }
 
+    // -- todo rotateLeft
+
+    @Test
+    public void shouldRotateLeftOnEmpty() {
+        assertThat(empty().rotateLeft(1)).isEqualTo(empty());
```
Comment (ruslansennov) [inline vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java]: isSameAs

File: vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java [hunk 2]
```diff
@@ -1489,6 +1489,86 @@ public void shouldCreateReverseIteratorOfNonEmpty() {
...
+
+    @Test
+    public void shouldRotateLeftForPositiveGreaterThatLen() {
+        assertThat(of(1, 2, 3, 4, 5).rotateLeft(5 + 2)).isEqualTo(of(3, 4, 5, 1, 2));
+    }
+
+    @Test
+    public void shouldRotateLeftForPositiveModuloLen() {
+        Seq<Integer> seq = of(1, 2, 3, 4, 5);
+        assertThat(seq.rotateLeft(seq.length() * 3)).isSameAs(seq);
```
Comment (ruslansennov) [inline vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java]: isSameAs

File: vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java [hunk 3]
```diff
@@ -1489,6 +1489,86 @@ public void shouldCreateReverseIteratorOfNonEmpty() {
...
+    }
+
+    // -- rotateRight
+
+    @Test
+    public void shouldRotateRightOnEmpty() {
+        assertThat(empty().rotateRight(1)).isEqualTo(empty());
```
Comment (ruslansennov) [inline vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java]: isSameAs

File: vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java [hunk 4]
```diff
@@ -1489,6 +1489,86 @@ public void shouldCreateReverseIteratorOfNonEmpty() {
...
+
+    @Test
+    public void shouldRotateRightForPositiveGreaterThatLen() {
+        assertThat(of(1, 2, 3, 4, 5).rotateRight(5 + 2)).isEqualTo(of(4, 5, 1, 2, 3));
+    }
+
+    @Test
+    public void shouldRotateRightForPositiveModuloLen() {
+        Seq<Integer> seq = of(1, 2, 3, 4, 5);
+        assertThat(seq.rotateRight(seq.length() * 3)).isSameAs(seq);
```
Comment (ruslansennov) [inline vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java]: isSameAs

File: vavr/src/test/java/io/vavr/collection/CharSeqTest.java [hunk 5]
```diff
@@ -2935,6 +2935,86 @@ public void shouldReverseNonNil() {
         assertThat(CharSeq.of('1', '2', '3').reverse()).isEqualTo(CharSeq.of('3', '2', '1'));
     }
 
+    // -- rotateLeft
+
+    @Test
+    public void shouldRotateLeftOnEmpty() {
+        assertThat(CharSeq.empty().rotateLeft(1)).isEqualTo(CharSeq.empty());
```
Comment (ruslansennov) [inline vavr/src/test/java/io/vavr/collection/CharSeqTest.java]: isSameAs

File: vavr/src/test/java/io/vavr/collection/CharSeqTest.java [hunk 6]
```diff
@@ -2935,6 +2935,86 @@ public void shouldReverseNonNil() {
...
+    public void shouldRotateLeftForPositiveModuloLen() {
+        CharSeq seq = CharSeq.of('1', '2', '3', '4', '5');
+        assertThat(seq.rotateLeft(seq.length() * 3)).isSameAs(seq);
+    }
+
+    // -- rotateRight
+
+    @Test
+    public void shouldRotateRightOnEmpty() {
+        assertThat(CharSeq.empty().rotateRight(1)).isEqualTo(CharSeq.empty());
```
Comment (ruslansennov) [inline vavr/src/test/java/io/vavr/collection/CharSeqTest.java]: isSameAs

File: vavr/src/main/java/io/vavr/collection/Seq.java [hunk 7]
```diff
@@ -826,6 +826,28 @@ default int prefixLength(Predicate<? super T> predicate) {
     Iterator<T> reverseIterator();
 
     /**
+     * Circular rotates the elements elements by the specified distance to the left direction.
```
Comment (ruslansennov) [inline vavr/src/main/java/io/vavr/collection/Seq.java]: "elements elements"

File: vavr/src/main/java/io/vavr/collection/Seq.java [hunk 8]
```diff
@@ -826,6 +826,28 @@ default int prefixLength(Predicate<? super T> predicate) {
...
+     * <pre>{@code
+     * Seq.of(1, 2, 3, 4, 5).rotateLeft(2);//[3, 4, 5, 1, 2]
+     * }</pre>
+     *
+     * @return the rotated elements.
+     */
+    Seq<T> rotateLeft(int n);
+
+    /**
+     * Circular rotates the elements elements by the specified distance to the left direction.
```
Comment (ruslansennov) [inline vavr/src/main/java/io/vavr/collection/Seq.java]: * "elements elements"
* **right** direction

**Comment by valery1707** [comment]
@ruslansennov fixed

**Comment by ruslansennov** [comment]
thank you

**Pull request merged**